core/src/main/scala/fluflu/Messenger.scala

```diff
@@ -0,0 +1,40 @@
+package fluflu
+
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.{ ConcurrentLinkedQueue, ForkJoinPool }
+
+import scala.util.Try
+
+class Messenger(
+    conn: Connection,
+    letterQueue: ConcurrentLinkedQueue[Letter],
+    retryHandler: (Letter) => PartialFunction[Throwable, Unit]
+) extends Thread {
+
+  val pool = new ForkJoinPool(1)
+
+  def enqueue(letter: Letter): Unit = synchronized {
+    if (conn.isClosed) conn.connect()
+    letterQueue.add(letter)
+  }
+
+  override def run(): Unit = {
+    try {
+      while (true) {
+        if (!conn.isClosed) {
+          Option(letterQueue.poll()).map { letter =>
+            pool.submit(new Runnable {
+              override def run(): Unit =
+                Try(conn.write(letter.message)).recover(retryHandler(letter))
+            }).get()
+          }
+        }
+        Thread.sleep(50)
```

Comment:
As sleeping a thread never notifies the scheduler, it might be a good idea to use `java.concurrent.ManagedBlocker`.